File: app/src/main/java/unforgettable/azcs/me/unforgettable/Sentence.java
package unforgettable.azcs.me.unforgettable;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by abdulazizalawshan on 1/6/18.
 */

public class Sentence implements Parcelable {
    public static final Creator<Sentence> CREATOR = new Creator<Sentence>() {
        @Override
        public Sentence createFromParcel(Parcel in) {
            return new Sentence(in);
        }

        @Override
        public Sentence[] newArray(int size) {
            return new Sentence[size];
        }
    };
    private String sentence;
    private String key;

    public Sentence(String sentence, String key) {
        this.sentence = sentence;
        this.key = key;
    }

    protected Sentence(Parcel in) {
        sentence = in.readString();
        key = in.readString();
    }

    public String getSentence() {
        return sentence;
    }

    public void setSentence(String sentence) {
        this.sentence = sentence;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(sentence);
        dest.writeString(key);
    }
}

File: app/src/main/java/unforgettable/azcs/me/unforgettable/Sentence.kt
package unforgettable.azcs.me.unforgettable

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by abdulazizalawshan on 1/6/18.
 */

class Sentence : Parcelable {
    var sentence: String? = null
    var key: String? = null

    constructor() {}

    constructor(sentence: String, key: String) {
        this.sentence = sentence
        this.key = key
    }

    protected constructor(`in`: Parcel) {
        sentence = `in`.readString()
        key = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(sentence)
        dest.writeString(key)
    }

    companion object {
        val CREATOR: Parcelable.Creator<Sentence> = object : Parcelable.Creator<Sentence> {
            override fun createFromParcel(`in`: Parcel): Sentence {
                return Sentence(`in`)
            }

            override fun newArray(size: Int): Array<Sentence?> {
                return arrayOfNulls(size)
            }
        }
    }
}
